frontend/src/components/common/Notifications/index.tsx

```diff
@@ -0,0 +1,210 @@
+import { Notification } from '../../../lib/notification';
+import { createRouteURL } from '../../../lib/router';
+import { timeAgo } from '../../../lib/util';
+import { setUINotifications, updateUINotification } from '../../../redux/actions/actions';
+import { useTypedSelector } from '../../../redux/reducers/reducers';
+
+const useStyles = makeStyles((theme: Theme) => ({
+  notificationItem: {
+    '&.MuiMenuItem-root': {
+      borderBottom: `1px solid ${theme.palette.notificationBorderColor}`,
+      padding: '1rem',
+    },
+  },
+  notificationMessage: {
+    '&': {
+      wordBreak: 'break-word',
+      whiteSpace: 'normal',
+    },
+  },
+  root: {
+    '& .MuiPaper-root': {
+      width: '30vw',
+      minWidth: '300px',
+      maxHeight: '400px',
+    },
+  },
+  errorItem: {
+    '&': {
+      color: theme.palette.error,
+    },
+  },
+}));
+
+function prepareNotificationsList(
+  notifications: Notification[],
+  clickEventHandler: (notification?: Notification) => void
+) {
+  const { t } = useTranslation();
+  const classes = useStyles();
+  if (!notifications || notifications.length === 0) {
+    return (
+      <MenuItem onClick={() => clickEventHandler()} className={classes.notificationItem}>
+        {t(`frequent|You don't have any notifications right now`)}
+      </MenuItem>
+    );
+  }
+
+  function menuItemClickHandler(notification?: Notification) {
+    if (!notification) {
+      return;
+    }
+    notification.seen = true;
+    clickEventHandler(notification);
+  }
+
+  function prepareNotificationLinkComponent(notification: Notification) {
+    if (!notification.url) {
+      return notification.message;
+    }
+    return (
+      <Link component={RouterLink} to={notification.url}>
+        {notification.message}
+      </Link>
+    );
+  }
+
+  return notifications?.map((notification, index: number) => {
+    return (
+      <MenuItem
+        onClick={() => menuItemClickHandler(notification)}
+        key={`${notification}__${index}`}
+        className={classes.notificationItem}
+      >
+        <Grid container spacing={4}>
+          <Grid item direction="column" md={notification.seen ? 12 : 11}>
+            <Grid item md={12}>
+              <Typography style={{ fontWeight: notification.seen ? 'normal' : 'bold' }}>
+                {prepareNotificationLinkComponent(notification)}
+              </Typography>
+            </Grid>
+            <Grid item>{timeAgo(notification.date)} ago</Grid>
+          </Grid>
+          {!notification.seen && (
+            <Grid item md={1}>
+              <Badge variant="dot" color="error"></Badge>
+            </Grid>
+          )}
+        </Grid>
+      </MenuItem>
+    );
+  });
+}
+
+export default function Notifications() {
+  const classes = useStyles();
+  const [anchorEl, setAnchorEl] = useState(null);
+  const notifications = useTypedSelector(state => state.ui.notifications);
+  const dispatch = useDispatch();
+  const [events] = Event.useList();
```

Comment:
I get the feeling we shouldn't mix the events here and just have this logic handle Notifications in a pure way. But I don't know where I'd put the events' logic right now... So maybe we can keep this as a first version.